osu.Game.Rulesets.Catch.Tests/TestSceneHyperDashColouring.cs

```diff
@@ -68,6 +110,40 @@ public void TestFruitColourFallback()
             checkHyperDashFruitColour(skin, skin.HyperDashColour);
         }
 
+        private void checkHyperDashCatcherColour(ISkin skin, Color4 expectedCatcherColour, Color4? expectedEndGlowColour = null)
+        {
+            CatcherArea catcherArea = null;
+
+            AddStep("create hyper-dashing catcher", () =>
+            {
+                Child = setupSkinHierarchy(catcherArea = new CatcherArea
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    Scale = new Vector2(4f),
+                }, skin);
+
+                catcherArea.MovableCatcher.SetHyperDashState(2);
+                catcherArea.MovableCatcher.FinishTransforms();
+            });
+
+            AddAssert("catcher colour is correct", () =>
+                expectedCatcherColour == Catcher.DEFAULT_HYPER_DASH_COLOUR
+                    ? catcherArea.MovableCatcher.Colour == Catcher.DEFAULT_CATCHER_HYPER_DASH_COLOUR
+                    : catcherArea.MovableCatcher.Colour == expectedCatcherColour);
+
+            AddAssert("catcher trails colours are correct", () => catcherArea.OfType<Container<CatcherTrailSprite>>().Any(c => c.Colour == expectedCatcherColour));
+            AddAssert("catcher end-glow colours are correct", () => catcherArea.OfType<Container<CatcherTrailSprite>>().Any(c => c.Colour == (expectedEndGlowColour ?? expectedCatcherColour)));
```

Comment:
this is completely wrong

Reply:
I'm not quite sure how to test this correctly without directly exposing the containers...

I would say applying what @bdach suggested regarding grouping all trail containers in one `CatcherTrailsContainer` may help this by doing:
```csharp
catcherArea.OfType<CatcherTrailsContainer>().Single().HyperDashTrails == expectedColour
```

Reply:
Try it and see how it feels. Could work.

Reply:
After creating a public `CatcherTrailsDisplay` grouping all trails, I've decided to also move the trail-related logic into that component now that there's a chance to, for simplifying `Catcher`.